### ISSUE TITLE
Nonceminer not generate reward

### ISSUE THREAD

**Issue by mahesh1987-16** [colonelwatch/nonceMiner#27]
Which nonceminerdo I use to get good M/H speed?
I was quite happy.
I did that for 2 hours only.I was shocked.  I saw that he didn't reward even duino-coin one in 2 hours.
The official software of Duino Coin was generating both the duino cion
The nonceminer minor was working but it was not generating the duino coin.
I tried every way and searched a lot on the net.  Didn't get any reply about it.
Then I got very disappointed.  Please someone help regarding this.

( My english is a little week, so i have taken help of google translate in it. )

**Comment by colonelwatch** [comment]
nonceMiner offers a very fast implementation of DUCO-S1, but this doesn't translate to higher rewards in duino-coin. You would think it should, but reward policies are completely up to the developers because they control their server. Regarding your issue, I think the developers decided to effectively zero the reward for anything going as fast as nonceMiner enables.

nonceMiner does offers a couple tools that might help though: the `-i`, `-t`, and `-l` options. If you play with these, you might find a configuration that maximizes reward. You might have to step through some configurations that are slower, but you have a *lot* of headroom as you saw. I'll warn though that reward maximization is not a responsibility/promise of nonceMiner, so I can't really help you there. Maybe a good starting position is `-i MEDIUM -t 12 -l 0.5`? Again, no guarantees.

That said, I'd really recommend reviewing the current reward policies before you experiment. Here's a permalink to the reward policy [in the server source code](https://github.com/revoxhere/duino-coin/blob/e51ea161e4952e9a8a5fc355b8f932d5a6fd8287/Master_Server/Server.py#L339) as of 9/26/2021. There is also a maximum limit of [50 threads](https://github.com/revoxhere/duino-coin/blob/e51ea161e4952e9a8a5fc355b8f932d5a6fd8287/Master_Server/Server.py#L77) set by the developers as of 9/26/2021.

**Comment by mahesh1987-16** [comment]
I used the same parameters that you had given.It was working perfect, enjoyed it.I did many more experiments.  Most all were successful.
But as you told.  It has some term and condition like you can use only 50 threads. -l 0.5 And only this can be set.
Because if the zero point is more than 5, both the Duino coin not generated.
I was wondering one thing that you have written in this script.

> My laptop (Intel i7-8550U + UHD Graphics 630) averages about 60 MH/s. In other words, it's the fastest open-source DUCO-S1 > miner around. Here are some other figures as July 25, 2021:
> 
> Nvidia RTX 3070: 1.65 GH/s
> Nvidia GTX 1060 6GB: 684 MH/s
> Nvidia GTX 770: 343 MH/s

You can't use more than 50 thread and you can't use more than -l 0.5.

Then how did you get so many giga hash and megha hash from the graphic card?

Because as far as I have calculated, you will not be able to take this much speed according to the rules

Then what parameters did you use to take this speed.  Please tell it once on the exam too. 

( My english is a bit weak, so i have taken help of google translate.)

**Comment by colonelwatch** [comment]
Yes, your calculations are right. I was only able to achieve those speeds if I neglected reward. Actually running your computer like  I posted generates zero reward.

(Read on if you want, but your question is answered in the above)

Its one of those things that make you think really. 50 workers limited at 0.5 MH/s each means a user can only hash at 25 MH/s at MEDIUM difficulty. And its only 50 MH/s at NET difficulty. But there can be so much more power available in an average computer, so it makes you wonder why you just can't open a new account to get around that. Lo and behold, that's a Sybil attack, exploits by using multiple online identities.

There are many ways to have Sybil resistance. Proof of work (or stake) is one of the only decentralized ways to do it, one of the reasons why Bitcoin was so revolutionary. duino-coin depends on identity verification (IP, emails, photo proof, etc). That works too, but it also means duino-coin may never truly decentralize. I never knew about this before I started this project, but its very interesting stuff.
